lib/LANraragi/Utils/Vips.pm

```diff
@@ -0,0 +1,281 @@
+} else {
+    # Dummy functions if libvips is not loaded
+    *vips_init = sub { die "libvips is not loaded. Cannot call vips_init." };
+    *vips_error_buffer = sub { die "libvips is not loaded. Cannot call vips_error_buffer." };
+    *vips_error_clear = sub { die "libvips is not loaded. Cannot call vips_error_clear." };
+    *vips_image_new_from_file = sub { die "libvips is not loaded. Cannot call vips_image_new_from_file." };
+    *vips_jpegsave = sub { die "libvips is not loaded. Cannot call vips_jpegsave." };
+    *vips_image_write_to_buffer = sub { die "libvips is not loaded. Cannot call vips_image_write_to_buffer." };
+    *g_object_unref = sub { die "libvips is not loaded. Cannot call g_object_unref." };
+    *g_free = sub { die "libvips is not loaded. Cannot call g_free." };
+    *vips_image_new_from_buffer = sub { die "libvips is not loaded. Cannot call vips_image_new_from_buffer." };
+    *vips_image_get_width = sub { die "libvips is not loaded. Cannot call vips_image_get_width." };
+    *vips_image_get_height = sub { die "libvips is not loaded. Cannot call vips_image_get_height." };
+    *vips_image_get_bands = sub { die "libvips is not loaded. Cannot call vips_image_get_bands." };
+    *vips_black = sub { die "libvips is not loaded. Cannot call vips_black." };
+    *vips_crop = sub { die "libvips is not loaded. Cannot call vips_crop." };
+    *vips_colourspace = sub { die "libvips is not loaded. Cannot call vips_colourspace." };
+    *vips_insert = sub { die "libvips is not loaded. Cannot call vips_insert." };
+    *vips_pngsave = sub { die "libvips is not loaded. Cannot call vips_pngsave." };
+    *vips_thumbnail_buffer_plain = sub { die "libvips is not loaded. Cannot call vips_thumbnail_buffer_plain." };
+    *vips_thumbnail_buffer = sub { die "libvips is not loaded. Cannot call vips_thumbnail_buffer." };
+}
+
+# Define VipsInterpretation enum values
+use constant VIPS_INTERPRETATION_GREY16 => 12; # VIPS_INTERPRETATION_GREY16
+use constant VIPS_INTERPRETATION_B_W => 17; # VIPS_INTERPRETATION_B_W
+use constant VIPS_SIZE_FORCE => 3; # VIPS_SIZE_FORCE
+use constant VIPS_ALIGN_CENTRE => 0; # VIPS_ALIGN_CENTRE
+use constant VIPS_INTERESTING_CENTRE => 1;
+
+my $initialized = 0;
+
+
+
+sub init ($program_name) {
+    if (!$VIPS_LOADED) {
+        warn "libvips is not loaded. Cannot initialize Vips.\n";
+        return 0; # Indicate failure
+    }
+    if (!$initialized) {
+        my $ret = vips_init($program_name);
+        if ($ret != 0) {
+            my $error = error_buffer();
+            clear_error();
+            die "Error initializing libvips: $error";
+        }
+        $initialized = 1;
+    }
+    return 1;
+}
+
+sub error_buffer () {
+    return vips_error_buffer();
+}
+
+sub clear_error () {
+    vips_error_clear();
+}
+
+sub new_from_file ($filename) {
+    my $image = vips_image_new_from_file($filename, undef);
+    if (!$image) {
+        my $error = error_buffer();
```

Comment:
I imagine this binding code was auto-generated, but it's probably worth it to refactor the `error_buffer` read/clear/die dance into a single sub that can be called in one line in those methods.